components/kserve/kserve.go

```diff
@@ -131,6 +132,12 @@ func (k *Kserve) ReconcileComponent(ctx context.Context, cli client.Client,
 				return err
 			}
 		}
+		extraParamsMap := map[string]string{
```

Comment:
This block, most likely, also needs to be applied to model mesh side, as we share odh-model-controller.

Reply:
I think this may be conflicting as in fact we are installing the same odh model controller manifests for both kserve and modelmesh components

Reply:
@lburgazzoli @israel-hdez 
Can you please advise me on what the suggested request is? I'm trying understand the ask.

Reply:
lets clarify something first:
this NIM flag is only used by kserve but not modelmesh, right?

if so we have an empty value key in  odh-model-controller's params.env, e.g
`nim-state:`

then the value will only be added by kserve to either Managed or Removed.
even the same params.env is shared by kserve and modelmesh, wont be impact for modelmesh.

is that correct?

Reply:
We talked a little bit about this on slack, but:

> this NIM flag is only used by kserve but not modelmesh, right?
> 

AFAIK, yes.

> if so we have an empty value key in odh-model-controller's params.env, e.g
> nim-state:
> 
> then the value will only be added by kserve to either Managed or Removed.
> even the same params.env is shared by kserve and modelmesh, wont be impact for modelmesh.
> 
> is that correct?

Yes. Strictly speaking, AFAIK, KServe and ModelMesh controllers won't use this flag because the NIM integration would be implemented in odh-model-controller.

I guess the motivation for it being under KServe is that the NIM integration is built on top of KServe. So, it is important that the feature flag is properly set as on/off in odh-model-controller depending on what is enabled in the DSC.

Reply:
After some clarification, the best options would be to include the NIM from the `KServer` component also when rendering the odh-model-controller manifests for the `ModelMesh` component:

```yaml
nim-state: {{ .Spec.Components.Kserve.NIM.ManagementState }}
# if knowing the management state of Kserve & ModelMesh is relevant, the following
# parameters can be included too
kserve-management-state: {{ .Spec.Components.Kserve.ManagementState }}
modelmesh-management-state: {{ .Spec.Components.ModelMesh.ManagementState }}
```

This means that both the `Kserve` and `ModelMesh` component reconcile method should be amended.

Note that this would probably require to:
- change the signature of the [Component interface ](https://github.com/opendatahub-io/opendatahub-operator/blob/incubation/components/component.go#L85-L86) to accept a `client.Object `as owner
- access to the field of the sibling component should happen using the unstructured type since it is not possible to cast the `owner` param to a `DataScienceCluster` interface (due to circular references)